src/cache.rs

```diff
@@ -150,16 +150,20 @@ impl Cache {
         let client = builder
             .build()
             .context("Could not instantiate HTTP client")?;
-        let mut resp = client.get(&self.url).send()?;
+        let mut resp = client
+            .get(&self.url)
+            .send()?
```

Comment:
We don't provide any `context` here, but the errors are pretty clear so we can leave it like this :)

Nice find with `error_for_status`!